rotkehlchen/tests/api/test_assets.py

```diff
@@ -607,6 +611,45 @@ def test_search_assets(rotkehlchen_api_server):
     result = assert_proper_response_with_result(response)
     assert {asset['chain'] for asset in result} == {'matic', 'optimism', 'ethereum', 'arbitrum', 'binance'}  # noqa: E501
 
+    # check that using chain filter works.
+    response = requests.post(
+        api_url_for(
+            rotkehlchen_api_server,
+            'assetssearchresource',
+        ),
+        json={
+            'value': 'DAI',
+            'search_column': 'symbol',
+            'limit': 50,
+            'chain': 'ethereum',
+            'order_by_attributes': ['name'],
+            'ascending': [True],
+        },
+    )
+    result = assert_proper_response_with_result(response)
+    assert 50 >= len(result) > 10
+    assert all(['ethereum' == entry['chain'] for entry in result])
+    for entry in result:
+        assert 'DAI' in entry['symbol']
```

Comment:
here you are iterating twice the results. Personally I would prefer a unique for loop but if you want to use the `all` statement you can do

```suggestion
    assert all(['ethereum' == entry['chain'] and 'DAI' in entry['symbol'] for entry in result])
```